kml/serdes.cpp

```diff
@@ -306,11 +306,13 @@ void SaveCategoryExtendedData(KmlWriter::WriterWrapper & writer, CategoryData co
     writer << "</mwm:author>\n";
   }
 
-  if (categoryData.m_lastModified != Timestamp())
-  {
-    writer << indent << "<mwm:lastModified>" << TimestampToString(categoryData.m_lastModified)
-           << "</mwm:lastModified>\n";
+  Timestamp lastModifiedTime;
+  if (categoryData.m_lastModified != Timestamp()) {
+    lastModifiedTime = categoryData.m_lastModified;
+  } else {
+    lastModifiedTime = kml::TimestampClock::now();
   }
+  writer << indent << "<mwm:lastModified>" << TimestampToString(lastModifiedTime) << "</mwm:lastModified>\n";
```

Comment:
I suspect that this change breaks all test's constrains. We do Save X, but Read Y.
We change the inner state in the Save function which is not good.

I still haven't read all the PR descriptions, but if this is the real error case, we _should assign_ a valid timestamp somewhere else: e.g. in the Import, Create, Modify functions, but not in Save.

Reply:
> @cyber-toad Am I right that m_lastModified for lists (categories) is not initialized now neither for KML nor for GPX imports? If yes, can you please help with a fix?

@biodranik yes, it is not used in parser an reset to current time is done later in UpdateLastModifiedTime. So we can have it parsed there if it is in file, but if we will initialize it to default (current) time in parser we will have different result for two executions of parser on the same source, that will make it more difficult to test (we will need to pass current time for test from test code). Or there is a better place than parser for defaulting?

Reply:
Does it make sense to add `Platform::GetFileModificationTime` in the same way as `Platform::GetFileCreationTime`, and use it during the import? Or maybe even rename it to `GetFileTime` and pass an enum `ECreationTime` and `EModificationTime` as a parameter?

Reply:
I'm not sure that these values are stable, for example:

```
~/dev/organicmaps/data/gpx_test_data$ stat route.gpx
  File: route.gpx
...
Access: 2024-03-08 10:00:05.705673721 +0100
Modify: 2024-01-14 15:45:23.712451946 +0100
Change: 2024-01-14 15:45:23.712451946 +0100
 Birth: 2024-01-14 15:45:23.712451946 +0100
```

So birth/change/modify were set during git pull, so will be changed if another dev made a checkout and tests fail?

Reply:
The test can import a file without that field, and compare that field to manually retrieved file modification date.

Reply:
Comparison with a modification time stored in the file should be easy - it should not change during the import. Then we can preserve the time for backups.

Reply:
@cyber-toad can you please help me to fix this importing issue?

Reply:
@biodranik so you suggest to enrich all current test gpx/kml/kmb data with this timestamp to make existing tests stable and add a couple of new tests where this field is empty and test compares defaulted value with date of file extracted in tests?

@kirylkaveryn I can have a look on it, but I'm a bit busy now so can't provide any ETA. But I see that you extracted it to separate task so will have a look once there is a time.